[CI] Seed specific failure of RemoveCorruptedShardDataCommandTests.testCorruptedTranslog

`RemoveCorruptedShardDataCommandTests.testCorruptedTranslog` fails reproducibly on the 7.x branch (confirmed using commit bb91291273f6c89426d43174177c0dce8d8cadb9) with this repro command:

```
./gradlew ':server:test' --tests "org.elasticsearch.index.shard.RemoveCorruptedShardDataCommandTests.testCorruptedTranslog" \
  -Dtests.seed=73D8350A7F9C1987 \
  -Dtests.security.manager=true \
  -Dtests.locale=sl \
  -Dtests.timezone=Europe/Brussels \
  -Dcompiler.java=12 \
  -Druntime.java=12
```

and this error:

```
  2> java.lang.AssertionError: minTranslogGen [11] is higher than generation [10]
        at __randomizedtesting.SeedInfo.seed([73D8350A7F9C1987:41A0283FE58FDB57]:0)
        at org.elasticsearch.index.translog.Checkpoint.<init>(Checkpoint.java:100)
        at org.elasticsearch.index.translog.TestTranslog.corruptRandomTranslogFile(TestTranslog.java:99)
        at org.elasticsearch.index.translog.TestTranslog.corruptRandomTranslogFile(TestTranslog.java:72)
        at org.elasticsearch.index.shard.RemoveCorruptedShardDataCommandTests.testCorruptedTranslog(RemoveCorruptedShardDataCommandTests.java:220)
```

If you change the seed then it doesn't fail.

The CI build where this happened was https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java12,nodes=general-purpose/371/console

--- a/server/src/test/java/org/elasticsearch/index/translog/TestTranslog.java
+++ b/server/src/test/java/org/elasticsearch/index/translog/TestTranslog.java
@@ -92,11 +92,21 @@ public class TestTranslog {
                 // if we crashed while rolling a generation then we might have copied `translog.ckp` to its numbered generation file but
                 // have not yet written a new `translog.ckp`. During recovery we must also verify that this file is intact, so it's ok to
                 // corrupt this file too (either by writing the wrong information, correctly formatted, or by properly corrupting it)
-                final Checkpoint checkpointCopy = LuceneTestCase.usually(random) ? checkpoint
-                        : new Checkpoint(checkpoint.offset + random.nextInt(2), checkpoint.numOps + random.nextInt(2),
-                            checkpoint.generation + random.nextInt(2), checkpoint.minSeqNo + random.nextInt(2),
-                            checkpoint.maxSeqNo + random.nextInt(2), checkpoint.globalCheckpoint + random.nextInt(2),
-                            checkpoint.minTranslogGeneration + random.nextInt(2), checkpoint.trimmedAboveSeqNo + random.nextInt(2));
+                final Checkpoint checkpointCopy;
+                if (LuceneTestCase.usually(random)) {
+                    checkpointCopy = checkpoint;
+                } else {
+                    long newTranslogGeneration = checkpoint.generation + random.nextInt(2);
+                    long newMinTranslogGeneration = Math.min(newTranslogGeneration, checkpoint.minTranslogGeneration + random.nextInt(2));
+                    long newMaxSeqNo = checkpoint.maxSeqNo + random.nextInt(2);
+                    long newMinSeqNo = Math.min(newMaxSeqNo, checkpoint.minSeqNo + random.nextInt(2));
+                    long newTrimmedAboveSeqNo = Math.min(newMaxSeqNo, checkpoint.trimmedAboveSeqNo + random.nextInt(2));
+
+                    checkpointCopy = new Checkpoint(checkpoint.offset + random.nextInt(2), checkpoint.numOps + random.nextInt(2),
+                        newTranslogGeneration, newMinSeqNo,
+                        newMaxSeqNo, checkpoint.globalCheckpoint + random.nextInt(2),
+                        newMinTranslogGeneration, newTrimmedAboveSeqNo);
+                }
                 Checkpoint.write(FileChannel::open, unnecessaryCheckpointCopyPath, checkpointCopy,
                     StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW);
 

